core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,444 @@
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.os.Environment
+import android.text.Editable
+import android.text.TextWatcher
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import android.widget.Toast
+import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
+import androidx.fragment.app.DialogFragment
+import kotlinx.android.synthetic.main.dialog_add_note.add_note_edit_text
+import kotlinx.android.synthetic.main.dialog_add_note.add_note_text_view
+import kotlinx.android.synthetic.main.layout_toolbar.toolbar
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.coreComponent
+import org.kiwix.kiwixmobile.core.CoreApp.Companion.instance
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.core.extensions.toast
+import org.kiwix.kiwixmobile.core.reader.ZimReaderContainer
+import org.kiwix.kiwixmobile.core.utils.SharedPreferenceUtil
+import org.kiwix.kiwixmobile.core.utils.dialog.AlertDialogShower
+import org.kiwix.kiwixmobile.core.utils.dialog.KiwixDialog
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+/**
+ * Created by @author Aditya-Sood (21/05/19) as a part of GSoC 2019
+ *
+ * AddNoteDialog extends DialogFragment and is used to display the note corresponding to a
+ * particular article (of a particular zim file/wiki/book) as a full-screen dialog fragment.
+ *
+ * Notes are saved as text files at location: "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+ */
+
+// constant
+const val DISABLE_ICON_ITEM_ALPHA = 130
+const val ENABLE_ICON_ITEM_ALPHA = 255
+
+class AddNoteDialog : DialogFragment() {
+  private var zimFileName: String? = null
+  private var zimFileTitle: String? = null
+  private var articleTitle: String? = null
+
+  // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+  private lateinit var articleNoteFileName: String
+  private var noteFileExists = false
+  private var noteEdited = false
+
+  private lateinit var root: View
+
+  // Keeps track of state of the note (whether edited since last save)
+  // Stores path to directory for the currently open zim's notes
+  private var zimNotesDirectory: String? = null
+
+  @Inject
+  lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @Inject
+  lateinit var alertDialogShower: AlertDialogShower
+  private val saveItem by lazy { toolbar.menu.findItem(R.id.save_note) }
+  private val shareItem by lazy { toolbar.menu.findItem(R.id.share_note) }
+  private val deleteItem by lazy { toolbar.menu.findItem(R.id.delete_note) }
+  override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
there should be a space between the fields and the methods

Reply:
done